src/test/test-dirent-util.c

```diff
@@ -0,0 +1,116 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include <dirent.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <sys/stat.h>
+
+#include "alloc-util.h"
+#include "dirent-util.h"
+#include "path-util.h"
+#include "stat-util.h"
+#include "string-util.h"
+#include "tests.h"
+
+static void test_dirent_ensure_type(void) {
+        struct dirent de;
+        int result;
+
+        de.d_type = DT_UNKNOWN;
```

Comment:
Rather than mocking dirent, create the equivalent directory and files, see src/test/test-fs-util.c for an example on how to do it

Reply:
@bluca Thanks for the review.

I went through the test-fs-util.c and I have some follow up question. 

-> I could not figure out, how to query create a files which is of type UNKNOWN. I tried to look for similar functions like symlink, mknod but, could not find one to create the type UNKNOWN. 

-> Also, when we create a file/directory, we still need to manually set `de.d_name = nameoffile` and also `de.d_type = typeof hedirectory` which is similar to the setting `de.d_type = DT_UNKNOWN/DT_REG/...`. Following up with this, when we create a file, if we set `de.d_name = nameoffile`, how do we set `de.d_type` for the file. 

Thank you.

Reply:
DT_UNKNOWN is generated only by some file systems, most will actually initialize the value properly.

i'd always start with struct dirent returned by the kernel itself, but if this does not work for all relevant cases then initialize the fields (or patch the fields) in your code instead. that's totally ok, if it's not possible (or not easy) to get the data reasonably from the kernel.

Reply:
thanks @poettering , returnning by the kernel, do you meant, `de.d_type = DT_UNKNOWN/DT_REG` ?, I can go ahead to initlalise using DT_UNKNOWN right for this case ?